Fhir converter

AkRampersad opened the pull request:
This is the initial FHIR converter pull request. The module produces an appropriate list of dictionaries where each dictionary has  CHROM, POS, REF, and ALT filled in (as long as the transcript is translatable/can be converted without a web call). I get an error 

  File "C:\Users\sr116\AppData\Local\Programs\Python\Python311\Lib\site-packages\oakvar\lib\base\master_converter.py", line 852, in run
    lines_data, immature_exit = main_converter.get_variant_lines(input_path, num_pool, start_line_no, batch_size)
    ^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: too many values to unpack (expected 2)

AkRampersad committed:
initial fhir converter push

fhir converter that validates if a JSON is a FHIR file and creates a list of dicts that currently have the reference and alternate base.

Need to add chromosome position and number.

AkRampersad committed:
Chromosome conversion and position

Chromosome conversion and position information are now acquired and converted properly.

If there is no chromosome present in txt file the accession number is used for now.

AkRampersad committed:
Update fhir-converter.py

converter produces proper dict regardless of if the information is found in a single variant resource or split among multiple resources (such as variant + molecular consequence).

Will not produce a dict/variant line if the variant transcript is not found in the gene_43_10000.sqlite (gencode mapper sqlite file) or in GRCh38_RefSeq2UCSC.txt. 

The converter will automatically select the newest version of the gencode sqlite file (done with lines 31-48)

AkRampersad committed:
Update fhir-converter.py

minor type casting change to fix error.

rkimoakbioinformatics commented:
`immature_exit` (which is bool) should be returned. Thus, at the end of get_variant_lines, it should be `return lines, immature_exit`. `immature_exit` tells whether the end of file has been reached or not.

AkRampersad committed:
make_total_lines and refactor

refactored the code to make code parsing and valuestring extraction separate from make_total_lines

self total_lines and lines are being produced correctly but I get a string indexing error:

`err.fhir-converter      0:1     string indices must be integers, not 'str'

AkRampersad committed:
FHIR-converter MVP

This is push has a working FHIR-converter module.

If a FHIR doc is converted into a OakVar sqlite then back into a FHIR document the given and produced FHIR files are the same. The only differences are the UUIDs. 

summation: FHIR files produced and FHIR files read are the same

AkRampersad committed:
Merge branch 'rkimoakbioinformatics:master' into fhir-converter

AkRampersad committed:
Update fhir-converter.yml

updated yml for FHIR-converter MVP

AkRampersad committed:
Update fhir-converter.md

updated .md file for FHIR-converter 1.0.0

rkimoakbioinformatics commented:
I get `[Errno 2] No such file or directory: 'None\\GRCh38_RefSeq2UCSC.txt'`. This file is missing?

AkRampersad committed:
Added Chromosome Mapping file 

Added Chromosome Mapping obtained from (https://github.com/dpryan79/ChromosomeMappings/blob/master/GRCh38_RefSeq2UCSC.txt)

rkimoakbioinformatics committed:
module dir fix

rkimoakbioinformatics commented:
Commited a minor module dir fix. Besides that:

>ov run exampleinput -t fhir # This produces exampleinput.sqlite
>ov run exampleinput__s0.primary.fhir.json # This produces exampleinput__s0.primary.fhir.json.sqlite

Comparing the two sqlite files, two variants from sample s0 are missing: One with position 39623 (TG deletion) and the other one with position 41617436 (C to G change). The second command produces the crv file without those two variants. Please find why the difference.

AkRampersad committed:
Update fhir-converter.py

implemented a method of obtaining chromosome number if the LOINC code is provided
